source/module_elecstate/elecstate.cpp

```diff
@@ -16,38 +16,49 @@ const double* ElecState::getRho(int spin) const
     return &(this->charge->rho[spin][0]);
 }
 
-void ElecState::fixed_weights(const std::vector<double>& ocp_kb)
+void ElecState::fixed_weights(
+   const std::vector<double>& ocp_kb,
+   const int &nbands,
+   const double &nelec)
 {
 
-    int num = 0;
-    num = this->klist->get_nks() * GlobalV::NBANDS;
+    assert(nbands>0);
+    assert(nelec>0.0);
+
+    const double ne_thr = 1.0e-5;
```

Comment:
I am a little curious about the threshold determining whether two numbers are equal here. It is 1e-5 now but why? I guess there must be developers directly use even == operator between two float numbers but, I think "the threshold determining two float numbers are equal" should be unified, somehow, how do you think?

Reply:
at very high temperatures, the distribution of electrons follow the Fermi-Dirac distribution rule. In that case, it is impossible to use "==" otherwise the occupied states will be too many to calculate